admin/test/features/step_definitions/manage_check_forms_steps.rb

```diff
@@ -30,8 +30,7 @@
 end
 
 Then(/^it should be added to the list of forms$/) do
-  sleep 2 # need to address this sleep , test fails randomly with out this
-  expect(manage_check_forms_page.available_checks.rows.count).to eql(@current_form_count + 1)
+  wait_until{expect(manage_check_forms_page.available_checks.rows.count).to eql(@current_form_count + 1)}
```

Comment:
@shekss I have reworked this to the following : 
wait_until{(manage_check_forms_page.available_checks.rows.count).eql?(@current_form_count + 1)}

I don't think having a expect after this is of any use as if the above fails it will throw a timeout error and if it passes it will pass the expect so we can treat the wait_until as a dynamic waiting assertion.